crates/typst-macros/src/elem.rs

```diff
@@ -977,27 +977,27 @@ fn create_fields_impl(element: &Elem) -> TokenStream {
                 }
             }
 
-            fn field(&self, id: u8) -> Option<#foundations::Value> {
-                let id = Fields::try_from(id).ok()?;
+            fn field(&self, id: u8) -> Result<#foundations::Value, #foundations::FieldAccessError> {
+                let id = Fields::try_from(id)?;
                 match id {
                     #(#field_arms,)*
-                    _ => None,
+                    _ => Err(#foundations::FieldAccessError::Internal),
```

Comment:
The `None` in these functions can also indicate that the field doesn't exist because depending on the function the matchers above aren't exhausive. If you try `#hide[].hidden` for instance, then you'll get the "this is a bug" message. I think we can just return `MissingField` and remove `Internal`.

Reply:
I’ve fixed it for `Fields::{field, field_with_styles, field_from_styles}`, but I’m not sure whether this can happen for `impl TryFrom<u8> for Fields`.

Reply:
It can't, this would indicate a bug. I'm okay with keeping the variant.